snuba/query/processors/tagsmap.py

```diff
@@ -136,6 +145,29 @@ def process_query(self, query: Query, request_settings: RequestSettings) -> None
             if not apply_optimization:
                 return
 
+        # Do not use flattened tags if tags are being unpacked anyway. In that case
+        # using flattened tags only implies loading an additional column thus making
+        # the query heavier and slower
+        if query.get_arrayjoin_from_ast() and self.__find_tags(
+            query.get_arrayjoin_from_ast()
+        ):
+            return
+        if query.get_groupby_from_ast():
+            for expression in query.get_groupby_from_ast():
+                if self.__find_tags(expression):
+                    return
+        if query.get_having_from_ast() and self.__find_tags(
+            query.get_having_from_ast()
+        ):
+            return
```

Comment:
I'm curious, does this ever get used?

Reply:
If you are talking about the HAVING conditions, yes, discoverV2 has it